### ISSUE TITLE
Be able to edit an incomplete submission from the CP even if all required fields are not completed

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#507]
**What are you trying to do?**
Incomplete submissions are like drafts. Sometimes, you would like to complete them in multiple times, before to submit them.
This could be the case for very long multi-pages forms.

For some reasons, like a support request, a user can ask the admin to complete its form submission which is still not submitted and still incomplete.

This is currently not possible if the forms contains required field. On save the CP will list all fields that cannot be blank, blocking the ability to save at the current state.

Currently, if we're trying to edit the submission of a registered user, the only solution is to log as this user and edit the form on front-end. This is not the most convenient way, and sometimes this is not possible like the users are not registered.

**What's your proposed solution?**
Like for native Craft entries, it would be nice to have a _Save Draft_ feature, which save the current step of the submission without checking of some required fields are complete or not.

**Comment by engram-design** [comment]
I'm a little on the fence about this, as I don't believe CP users should be allowed to circumvent validation, which is there for a reason. Incomplete submissions are similar to drafts, but not the same. But I certainly see the use-case you've pointed out, where an admin might want to fill in _some_ fields, and save the submission.

I'm not too keen on the draft of submissions idea, mostly because it adds complexity and gets messy. How do we know which submission to restore to the user on the front end, if there are multiple drafts for a submission for instance.

I may very well introduce a "Save Draft" button that saves values without validating the submission.

**Comment by romainpoirier** [comment]
`I'm a little on the fence about this, as I don't believe CP users should be allowed to circumvent validation, which is there for a reason.`
You're right, but for an example there could be a complex multi-pages forms, where some data could be available for completion at a certain time, or could be filled later because of a very time consuming fill-out. Right now, if the required fields of the next pages are not filled in, it's not possible to edit the submission like a draft because of the remaining blank required fields. Not being able to do this prevent to change some basic settings, like the status of the submission: `draft`, `complete`, `in progress`, `selected`...

`But I certainly see the use-case you've pointed out, where an admin might want to fill in some fields, and save the submission.`
Example: the file the front-end user is trying to submit is too large, so the admin has to upload it from the FTP, then index the asset and attach it to the submission.

`How do we know which submission to restore to the user on the front end, if there are multiple drafts for a submission for instance.`
In my case, when I'm trying to edit a submission for a project, I'm loading the first submission using `.one()` based of the value of the `email()` field using `currentUser.email` (as the form is showing up in a restricted area of the website).

`I may very well introduce a "Save Draft" button that saves values without validating the submission.`
Perfect! That would be the solution. Do you know approximately when this feature could be added?

**Comment by engram-design** [comment]
@romainpoirier No strict ETA sorry, but probably in the next 1-2 weeks.

**Comment by engram-design** [comment]
Added for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.7",
  "...": "..."
}
```

Then run `composer update`.

**Comment by romainpoirier** [comment]
Perfect, now I can edit and `Save as draft` an incomplete submission. When I just click `Save`, I get the validation errors because of missing required fields, which I guess is the expected behaviour.

Do you still plan the ability to change the submission status even if it's still incomplete (still a draft)?
This would be very useful for multi-pages forms, for example to be able to add a custom status based on the current completion, or for some whatever other reasons...

**Comment by engram-design** [comment]
@romainpoirier At some point, (per https://github.com/verbb/formie/issues/508), but there's a lot to consider and plan for that. It won't be included for a few more versions.

**Comment by engram-design** [comment]
Fixed in [1.4.8](https://github.com/verbb/formie/releases/tag/1.4.8)
